hyperlink/_url.py

```diff
@@ -970,7 +950,9 @@ def from_text(cls, text):
                             raise URLParseError('port must not be empty')
                         raise URLParseError('expected integer for port, not %r'
                                             % port_str)
-        family, host = parse_host(host)
+        # _, host = parse_host(host)
+        if host:
+            host = host.lstrip('[').rstrip(']')
```

Comment:
You can now witness the new `_AUTHORITY_RE` action, with improved "bad host" detection. Adding a few more tests before I'd consider it final. (Note that this doesn't use regex IPv6 validation, just the host syntax)